### PR TITLE
fix(dataflow,scala): Correctly translate return when it's an expression

### LINKED ISSUE #4975: [scala] Functions do not work as taint sinks
**Describe the bug**
Simple functions do not work as taint syncs, even though they should.

**To Reproduce**
https://semgrep.dev/s/ezzer17:simple-function-sink-2
If you remove the `return`, it works

**Expected behavior**
Rule has one match

**What is the priority of the bug to you?**

- P1: important to fix or quite annoying

### PR DESCRIPTION
**Pull request by kopecs**
Fixes #4975.

The underlying issue that we current do not translate return from the generic AST to IL when `return` in the source language may be an expression (e.g., Rust, Scala). This PR adds a case to handle this.

PR checklist:

- [x] Documentation is up-to-date
- [x] Changelog is up-to-date
- [x] Change has no security implications (otherwise, ping security team)

Test plan: test included for both implict & explict return in Scala.

### CODE REVIEW

File: semgrep-core/tests/OTHER/rules/taint_scala_return.scala [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def no_ret(x: Int) : Int = {
```
Comment (mmcqd) [inline semgrep-core/tests/OTHER/rules/taint_scala_return.scala]: This test should go in `semgrep-core/tests/tainting_rules/scala`

**Review by mmcqd — APPROVED** [review]
Great job! Fixing tainting bugs isn't easy

File: semgrep-core/tests/tainting_rules/scala/return_expr.scala [hunk 2]
```diff
@@ -0,0 +1,9 @@
+def no_ret(x: Int) : Int = {
```
Comment (IagoAbal) [inline semgrep-core/tests/tainting_rules/scala/return_expr.scala]: No need to change, but just FYI `tests/tainting_rules` is a leftover from when `semgrep-core -rules` could not parse taint rules and we had a different switch for running them. Nowadays I try to put taint tests under `tests/OTHER/rules/` and I just prefix them with `taint_`.

**Review by IagoAbal — APPROVED** [review]
Very nice! Could you please add this https://semgrep.dev/s/ezzer17:simple-function-sink-2 (the example provided in #4975) also as a test case?

**Pull request merged**